examples/samples/SendersReceiversWithQueues.ts

```diff
@@ -0,0 +1,92 @@
+import { ReceiveMode, Namespace, generateUuid, QueueClient, SendableMessageInfo, OnMessage, ServiceBusMessage, OnError, MessagingError, delay, ReceiveHandler } from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+let ns: Namespace;
+
+async function sendMessages(queueClient: QueueClient): Promise<void> {
+  const data: { lastName: string; firstName: string }[] = [
+    { lastName: "Einstein", firstName: "Albert" },
+    { lastName: "Heisenberg", firstName: "Werner" },
+    { lastName: "Curie", firstName: "Marie" },
+    { lastName: "Hawking", firstName: "Steven" },
+    { lastName: "Newton", firstName: "Isaac" },
+    { lastName: "Bohr", firstName: "Niels" },
+    { lastName: "Faraday", firstName: "Michael" },
+    { lastName: "Galilei", firstName: "Galileo" },
+    { lastName: "Kepler", firstName: "Johannes" },
+    { lastName: "Kopernikus", firstName: "Nikolaus" }
+  ];
+
+  for (let index = 0; index < data.length; index++) {
+    const element = data[index];
+    const message: SendableMessageInfo = {
+      body: `${element.firstName} ${element.lastName}`,
+      label: "Scientist",
+      timeToLive: 2 * 60 * 1000, // 2 minutes
+      messageId: generateUuid()
+    };
+
+    console.log(`Sending ${message.body}`);
+    await queueClient.send(message);
+  }
+}
+
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+
+  // populate the queue with messages
+  await sendMessages(client);
+
+  let rcvHandler: ReceiveHandler;
+
+  // Using an object so that we can pass it around by reference
+  let received = 0;
+
+  // retrieve all the messages that were sent to the queue (10 messages)
```

Comment:
This sample doesnt assume that the queue was empty before sending the messages. Meaning, it is very much possible that the queue will have more than 10 messages in a real world scenario when the user is trying out this sample.

In this case, the current code doesnt really ensure that we process only 10 messages.
There are 2 places where we attempt to ensure this.

- Inside the onMessage callback where we call stop() if the count reached 10.
- Outside in the while loop where we wait for 30 seconds or 10 messages whichever comes first

The onMessage callback will be called until the stop() call which is asynchronous completes. Meaning that the 11th, 12th 13th messages will be marked as "complete".
Therefore to ensure that we process only 10 messages, we need to pull the check on the count up top. If count has reached 10, then we call stop() else we complete it.

The outer while loop will exit and close the client, once it sees the count >= 10 and is done sleeping as per the last delay() call. 

At this point, the next `onMessage` callback attempts to "complete" the incoming message and that fails with the below:

![image](https://user-images.githubusercontent.com/16890566/49822241-6d133400-fd31-11e8-8397-6da3e00cda77.png)

This is because we attempt to abandon a message if the user's onMessage handler throws an error.

There are 2 action items here:
- Update the `onMessage` callback to not complete any messages after the 10th message
- Dont attempt to abandon the message when user's onMessage handler throws an error for a closed reciever link.